server/v2/api/grpcgateway/server.go

```diff
@@ -76,11 +76,10 @@ func New[T transaction.Tx](
 	s.logger = logger.With(log.ModuleKey, s.Name())
 	s.config = serverCfg
 	mux := http.NewServeMux()
-	interceptor, err := newGatewayInterceptor[T](logger, s.GRPCGatewayRouter, appManager)
+	err := registerGatewayToMux[T](logger, mux, s.GRPCGatewayRouter, appManager)
```

Comment:
I feel like the arguments to this function should be rearranged.

Since the signature is `register Gateway to Mux`

Gateway should come before mux as an argument

Reply:
hmm i think this means the function name is misleading then, as we aren't really registering the gateway to the mux. its just provided as a fallback. ill rename it.